File: app/src/main/java/com/softsquared/examplesharedpreference/ListviewAdapter.java
package com.softsquared.examplesharedpreference;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;

public class ListviewAdapter extends BaseAdapter {

    ArrayList<Data> mArrayList;
    Context mContext;
    LayoutInflater mLayoutInflater;
    TextView mTitle,mContents;

    public ListviewAdapter(ArrayList<Data> mArrayList, Context mContext) {
        this.mArrayList = mArrayList;
        this.mContext = mContext;
        mLayoutInflater= (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public int getCount() {
        return mArrayList.size();
    }

    @Override
    public Data getItem(int position) {

        return mArrayList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View view, ViewGroup parent) {

        view= mLayoutInflater.inflate(R.layout.item_layout,parent,false);
        mTitle=view.findViewById(R.id.item_layout_title);
        mContents=view.findViewById(R.id.item_layout_contents);

        mTitle.setText(mArrayList.get(position).getTitle());
        mContents.setText(mArrayList.get(position).getContents());
        return view;
    }
}

File: app/src/main/java/com/softsquared/examplesharedpreference/ListviewAdapter.kt
package com.softsquared.examplesharedpreference

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import java.util.*
import kotlinx.android.synthetic.main.item_layout.view.*

class ListviewAdapter(var mArrayList: ArrayList<Data>,mContext: Context) : BaseAdapter() {
    var mLayoutInflater: LayoutInflater
    var mTitle: TextView? = null
    var mContents: TextView? = null
    override fun getCount(): Int {
        return mArrayList.size
    }

    override fun getItem(position: Int): Data {
        return mArrayList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, view: View?, parent: ViewGroup): View {
        var view = view
        view = mLayoutInflater.inflate(R.layout.item_layout, parent, false)

        view.item_layout_title.text=mArrayList[position].title
        view.item_layout_contents.text=mArrayList[position].contents
        return view
    }

    init {
        mLayoutInflater = mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }
}